### DISCUSSION TITLE
Prisma Implicit M2N relationship doesn't apply pagination on nested object ?

### DISCUSSION THREAD

**Discussion by jeromeSH26** [prisma/prisma#10932] [Q&A] [answered]
Hi, happy new year !

have a model with stores and articles. stores have many articles and articles many stores

As this is an implicit M2N, Prisma has generated automatically the Join table `_articlesTohypermarches_stores`

The max number of articles per store is 10K, but I want for example only the 10 first and not the whole 10K 
I tried to apply pagination on `articles` when querying the stores but the query sent to the `_articlesTohypermarches_stores` table has no `LIMIT 10` thus resulting in fetching the 10K articles...

Is my query wrong or should I change to explicit M2N relationship ?

here are the query I'm testing, the model and the the queries sent to POSTGRES by Prisma, as well as the PRISMA Logs that showing the pb.

```
const test = await prisma.hypermarches_stores.findMany({
    where: {
      id: {
        equals: 209,
      },
    },
    select: {
      stoDisplayDesc: true,
      articles: {
        take: 10, // <--- this is not taking into account
        skip: 1,
        select: {
          title: true,
        },
      },
    },
    take: 2,
  });



model articles {
  id                            Int                          @id(map: "PK_0a6e2c450d83e0b6052c2793334")  @default(autoincrement()) 
  title                         String?                         @default("") @db.VarChar(3000)
  prioriteId                    Int?                        
  isAlerte                      Boolean?                        @default(false)
  categorieId                   Int?                         
  publicationStartDate          DateTime?                       @default(dbgenerated("'2021-12-29 17:02:14.651'::timestamp without time zone")) @db.Timestamp(6)
  publicationEndDate            DateTime?                       @default(dbgenerated("'2021-12-29 17:02:14.651'::timestamp without time zone")) @db.Timestamp(6)
  hasAction                     Boolean?                        @default(false)
  briefDescription              String?                         @default("") @db.VarChar(3000)
  content                       String?                         @default("")
  contact1                      String?                         @default("") @db.VarChar(300)
  contact2                      String?                         @default("") @db.VarChar(300)
  author                        String?                         @default("") @db.VarChar(300)
  recordCreationDate            DateTime                        @default(now()) @db.Timestamp(6)
  recordUpdateDate              DateTime                        @default(now()) @db.Timestamp(6)
  ref_categories                ref_categories?                 @relation(fields: [categorieId], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "FK_278d87b271a80d56e5d6cc0f888")
  ref_priorites                 ref_priorites?                  @relation(fields: [prioriteId], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "FK_a55acd217d26e0d60f57b5f38f7")
  articles_documents            articles_documents[]
  fonctions                     ref_fonctions[]
  magasins                      hypermarches_stores[]
  metiers              ref_metiers[]
  perimetres           ref_perimetres[]
  poles                ref_poles[]
  services             ref_services[]
  articles_statuts_contribution articles_statuts_contribution[]

  @@index([id], map: "IDX_0a6e2c450d83e0b6052c279333")
}


model hypermarches_stores {
  id                             Int                   @id(map: "PK_7b1178bff8dc98cbd6d68fc2dd0") @default(autoincrement()) 
  stoEan                         String                   @unique(map: "UQ_7fddfa16362979011e27b4d0b21") @db.VarChar(300)
  articles              articles[]
  @@index([id], map: "IDX_7b1178bff8dc98cbd6d68fc2dd")
}
```

![image](https://user-images.githubusercontent.com/51024915/147945077-fc2b61e5-0dd8-42ab-b902-cf9a84388b24.png)

**Comment by jeromeSH26** [comment]
After comments in Stackoverflow [(),](https://stackoverflow.com/questions/70567139/prisma-implicit-m2n-relationship-doesnt-apply-pagination-on-nested-object?noredirect=1#comment124759704_70567139) let me precised : the result is correct in a way that only 10 records are at the end sent back by the query. 
But for a set of 250 stores, that each have 10K articles, the performances are very bad. Tooks around 5mns to get the result back.
It seems that prisma is splitting the query in 3. Get the ID of the stores, then get the link stores-articles in the join table, then extract all articles.
Then, I assume that prisma is reconcilating (merging I guess) these 3 queries results in a single object to sebd it back as a final result.
Not sure which sequence of this is taking so much time, as If I run the queries one by one in PGADMIN, each of them takes less than 200ms. So maybe the "reconciliation" step has to be optimized

**Comment by TasinIshmam** [comment, marked as answer]
Hey @jeromeSH26 

Thanks for bringing this up and creating an issue (#10949). If it's alright, I will close this discussion in favor of the linked issue as this is likely a bug.

**Reply by jeromeSH26** [reply]
ok no pb thks to look into this issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
